autoblocks/_impl/tracerv2/run.py

```diff
@@ -0,0 +1,263 @@
+            "message": message,
+            "traceId": trace_id,
+            "timestamp": timestamp,
+            "properties": merged_properties,
+        }
+
+        evaluations = []
+        try:
+            evaluations = await gather_with_max_concurrency(
+                max_evaluator_concurrency,
+                [
+                    self._evaluate_event(
+                        event=event_dict,
+                        evaluator=evaluator,
+                    )
+                    for evaluator in evaluators
+                ],
+            )
+        except Exception:
+            log.error("TODO: DO something here")
+
+        print(evaluations)
+        req = self._client.post(
+            url=INGESTION_ENDPOINT,
+            json={
+                "message": message,
+                "traceId": trace_id,
+                "timestamp": timestamp,
+                "properties": merged_properties,
+            },
+            headers=replay_headers,
+            timeout=self._timeout_seconds,
+        )
+        req.raise_for_status()
+        resp = req.json()
+        return SendEventResponse(trace_id=resp.get("traceId"))
+
+    async def async_send_event(
+        self,
+        message: str,
+        trace_id: Optional[str] = None,
+        span_id: Optional[str] = None,
+        parent_span_id: Optional[str] = None,
+        timestamp: Optional[str] = None,
+        properties: Optional[Dict] = None,
+        evaluators: List[BaseEvaluator] = [],
+        max_evaluator_concurrency: int = 5,
+    ) -> SendEventResponse:
+        # Do some assertions
+        await gather_with_max_concurrency(
+            max_evaluator_concurrency,
+            [
+                self._send_event_unsafe(
+                    message=message,
+                    trace_id=trace_id,
+                    span_id=span_id,
+                    parent_span_id=parent_span_id,
+                    timestamp=timestamp,
+                    properties=properties,
+                    evaluators=evaluators,
+                )
+            ],
+        )
+
+    def send_event(
+        self,
+        message: str,
+        # Require all arguments after message to be specified via key=value
+        *,
+        trace_id: Optional[str] = None,
+        span_id: Optional[str] = None,
+        parent_span_id: Optional[str] = None,
+        timestamp: Optional[str] = None,
+        properties: Optional[Dict] = None,
+        evaluators: List[BaseEvaluator],
+    ) -> SendEventResponse:
+        future = asyncio.run_coroutine_threadsafe(
+            self.async_send_event(
+                message=message,
+                trace_id=trace_id,
+                span_id=span_id,
+                parent_span_id=parent_span_id,
+                timestamp=timestamp,
+                properties=properties,
+                evaluators=evaluators,
+                max_evaluator_concurrency=5,
+            ),
+            loop,
+        )
+        future.result()
```

Comment:
i think we probably don't want to wait here, we should just fire and forget

Reply:
we'll need to make sure to attempt to wait for any outstanding futures on program exit though